File: leetcode_py/3653.py
from functools import reduce
import operator
from typing import List

class Solution:
    def xorAfterQueries(self, nums: List[int], queries: List[List[int]]) -> int:
      mod = 10 ** 9 + 7
      for ele in queries:
          l,r,k,v = ele
          for i in range(l,r + 1,k):
              nums[i] = (nums[i] * v) % mod
      return reduce(operator.xor,nums)
          


s = Solution()

s.xorAfterQueries([1,2,3],[[1,2,3,4],[4,5,6,7]])

File: leetcode_ts/3653.ts
function xorAfterQueries(nums: number[], queries: number[][]): number {
  const MOD = 1e9 + 7;
  for (const element of queries) {
    let [l, r, k, v] = element;
    for (let index = l; index <= r; index += k) {
      nums[index] = Number((BigInt(nums[index]) * BigInt(v)) % BigInt(MOD));
    }
  }
  let res: number = 0;
  // nums.forEach((element) => {
  //   res ^= element;
  // });
  for (const element of nums) {
    res ^= element;
  }
  return res;
}
